### PR TITLE
feat: Integrate local relationship writer DAO

### PR DESCRIPTION
**Pull request by zhixuanjia**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java [hunk 1]
```diff
@@ -252,4 +269,53 @@ public void testToAndFromJson() {
...
     assertNotNull(RecordUtils.toRecordTemplate(AspectFoo.class, EbeanLocalAccess.extractAspectJsonString(toJson)));
   }
+
+  @Test
+  public void testAddWithLocalRelationshipBuilder() throws URISyntaxException {
+    FooUrn fooUrn = makeFooUrn(1);
+    BarUrn barUrn1 = BarUrn.createFromString("urn:li:bar:1");
+    BarUrn barUrn2 = BarUrn.createFromString("urn:li:bar:2");
+    BarUrn barUrn3 = BarUrn.createFromString("urn:li:bar:3");
+    AspectFooBar aspectFooBar = new AspectFooBar().setBars(new BarUrnArray(new BarUrnArray(barUrn1, barUrn2, barUrn3)));
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java:279]: is this supposed to be two of the same constructor (new BarUrnArray(new BarUrnArray))

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java [hunk 2]
```diff
@@ -252,4 +269,53 @@ public void testToAndFromJson() {
...
+
+    _ebeanLocalAccessFoo.add(fooUrn, aspectFooBar, AspectFooBar.class, auditStamp);
+    _ebeanLocalAccessBar.add(barUrn1, new AspectFoo().setValue("1"), AspectFoo.class, auditStamp);
+    _ebeanLocalAccessBar.add(barUrn2, new AspectFoo().setValue("2"), AspectFoo.class, auditStamp);
+    _ebeanLocalAccessBar.add(barUrn3, new AspectFoo().setValue("3"), AspectFoo.class, auditStamp);
+
+    // Verify local relationships and entity are added.
+    EbeanLocalRelationshipQueryDAO ebeanLocalRelationshipQueryDAO = new EbeanLocalRelationshipQueryDAO(_server);
+    List<BelongsTo> relationships = ebeanLocalRelationshipQueryDAO.findRelationships(
+        BarSnapshot.class, new Filter().setCriteria(new CriterionArray()),
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java:290]: nit: we can make a static "EMPTY_FILTER" variable?

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java [hunk 3]
```diff
@@ -252,4 +269,53 @@ public void testToAndFromJson() {
...
+    List<EbeanMetadataAspect> aspects = _ebeanLocalAccessFoo.batchGetUnion(Collections.singletonList(key), 10, 0);
+
+    assertEquals(3, relationships.size());
+    assertEquals(1, aspects.size());
+  }
+
+  @Test
+  public void testAtomicityWithLocalRelationshipBuilder() throws URISyntaxException {
+    // Drop the entity table should fail add operation.
+    _server.createSqlUpdate("DROP TABLE metadata_entity_foo").execute();
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java:304]: maybe we need to add "IF EXISTS" here just in case?

**Review by jsdonn — CHANGES_REQUESTED** [review]

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java (hunk 3 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java:304]: "IF EXISTS" is left out on purpose. Test should fail if we can't drop the table.

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java (hunk 2 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java]: makes sense, updated

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java (hunk 1 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalAccessTest.java]: good catch, fixed

**Review by jsdonn — APPROVED** [review]
Thanks for addressing comments.

**Pull request merged**